eng/pipelines/templates/steps/prepare-pipelines.yml

```diff
@@ -0,0 +1,129 @@
+parameters:
+  - name: Repository
+    type: string
+    default: $(Build.Repository.Name)
+  - name: Prefix
+    type: string
+  - name: CIConventionOptions
+    type: string
+    default: ''
+  - name: UPConventionOptions
+    type: string
+    default: ''
+  - name: TestsConventionOptions
+    type: string
+    default: ''
+
+steps:
+  - template: install-pipeline-generation.yml
+
+  # This covers our public repos.
+  - ${{ if not(endsWith(parameters.Repository, '-pr'))}}:
+    - script: >
+        $(Pipeline.Workspace)/pipeline-generator/pipeline-generator
+        --organization https://dev.azure.com/azure-sdk
+        --project public
+        --prefix ${{parameters.Prefix}}
+        --devopspath "\${{parameters.Prefix}}"
+        --path $(System.DefaultWorkingDirectory)/sdk
+        --endpoint Azure
+        --repository ${{parameters.Repository}}
+        --convention ci
+        --agentpool Hosted
+        --branch refs/heads/master
+        --patvar PATVAR
+        --debug
+        ${{parameters.CIConventionOptions}}
+      env:
+        PATVAR: $(azuresdk-azure-sdk-devops-pipeline-generation-pat)
+    - script: >
+        $(Pipeline.Workspace)/pipeline-generator/pipeline-generator
+        --organization https://dev.azure.com/azure-sdk
+        --project internal
+        --prefix ${{parameters.Prefix}}
+        --devopspath "\${{parameters.Prefix}}"
+        --path $(System.DefaultWorkingDirectory)/sdk
+        --endpoint Azure
+        --repository ${{parameters.Repository}}
+        --convention up
+        --agentpool Hosted
+        --branch refs/heads/master
+        --patvar PATVAR
+        --debug
+        ${{parameters.UPConventionOptions}}
+      env:
+        PATVAR: $(azuresdk-azure-sdk-devops-pipeline-generation-pat)
+    - script: >
+        $(Pipeline.Workspace)/pipeline-generator/pipeline-generator
+        --organization https://dev.azure.com/azure-sdk
+        --project internal
+        --prefix ${{parameters.Prefix}}
+        --devopspath "\${{parameters.Prefix}}"
+        --path $(System.DefaultWorkingDirectory)/sdk
```

Comment:
Will this work for JS as it currently stands? I believe it might because it will just try to find all tests.yml file under this path, but we should at least figure out what currently happens for JS and if bad things happen we should add some conditions to avoid those bad things.

Reply:
I believe that it will work, because the convention searches all sub-folders.